src/elements/Header/Header.js

```diff
@@ -52,10 +52,13 @@ function Header(props) {
     return <ElementType {...rest} className={classes}>{children}</ElementType>
   }
 
-  if ((image && typeof image !== 'boolean') || (icon && typeof icon !== 'boolean')) {
```

Comment:
Cool, I'm on board with using the return of the factory to determine the markup.  Seems stronger to isolate that log in the factory's handling of the shorthand values.